app/modules/slack/webhooks_list.py

```diff
@@ -0,0 +1,294 @@
+                        "value": f"{end},{type}",
+                        "action_id": "next_page",
+                    }
+                ],
+            }
+        ]
+    else:
+        button_block = []
+    return button_block
+
+
+# Get and return all webhooks. Start is the index of the first webhook to show, end is the index of the last webhook to show.
+# Type is the type of webhooks to show (active, disabled, or all)
+def list_all_webhooks(
+    client, body, start, end, type, all_hooks, channel=None, update=False
+):
+    # each webhook consumes four blocks. max block size is 16. So, we need to divide the end index by 4 to get the actual number of webhooks to display
+    active_hooks = get_webhooks(all_hooks, "active")
+    active_hooks_list = get_webhooks_list(active_hooks)
+    active_button_block = get_webhooks_button_block("active", active_hooks_list, end)
+
+    # get the disabled webhooks and button block
+    disabled_hooks = get_webhooks(all_hooks, "disabled")
+    disabled_hooks_list = get_webhooks_list(disabled_hooks)
+    disabled_button_block = get_webhooks_button_block(
+        "disabled", disabled_hooks_list, end
+    )
+
+    if body.get("view", {"private_metadata": None}).get("private_metadata"):
+        private_metadata = json.loads(body["view"]["private_metadata"])
+    else:
+        private_metadata = {
+            "start": start,
+            "end": end,
+            "type": type,
+            "channel": channel,
+            "channel_name": body.get("channel_name", None),
+        }
+    header_active_hooks = f"There are currently {len(active_hooks)} enabled webhooks"
+    header_disabled_hooks = (
+        f"There are currently {len(disabled_hooks)} disabled webhooks"
+    )
+    if channel:
+        suffix = f" for channel:\n{private_metadata['channel_name']}"
+        header_active_hooks += suffix
+        header_disabled_hooks += suffix
+
+    # display the webhooks in the modal
+    blocks = {
+        "type": "modal",
+        "callback_id": "webhooks_view",
+        "title": {"type": "plain_text", "text": "SRE - Listing webhooks"},
+        "close": {"type": "plain_text", "text": "Close"},
+        "private_metadata": json.dumps(private_metadata),
+        "blocks": (
+            [
+                {
+                    "type": "header",
+                    "text": {
+                        "type": "plain_text",
+                        "text": header_active_hooks,
+                    },
+                },
+                {"type": "divider"},
+            ]
+            # this is used to traverse the list of webhooks for the pagination. Show the webhooks slice of array based on
+            # start and end index
+            + (
+                active_hooks_list[start:end]
+                if type == "active" or type == "all"
+                else active_hooks_list[0:MAX_BLOCK_SIZE]
+            )
+            # display the button block
+            + active_button_block
```

Comment:
Is this supposed to have a + sign next to it?

Reply:
At line 167, this is basically appending another block to the array of blocks. I haven't changed anything in these blocks, it just got moved in its own module